packages/web-components/@sfx/product/test/unit/product.test.ts

```diff
@@ -0,0 +1,241 @@
+
+      expect(result).to.have.length(2);
+      expect(result[0]).to.be.an.instanceOf(TemplateResult);
+      expect(result[1]).to.be.an.instanceOf(TemplateResult);
+    });
+  });
+
+  describe('urlWrap', () => {
+    const child = html`<span>foo</span>`;
+
+    it('should return the second argument wrapped in an "a" tag', () => {
+      const result = component.urlWrap('string', child);
+
+      expect(result).to.be.an.instanceof(TemplateResult);
+    });
+
+    it('should return the second argument', () => {
+      const result = component.urlWrap(undefined, child);
+
+      expect(result).to.equal(child);
+    });
+  });
+
+  describe('render', () => {
+    it('should return an instance of TemplateResult', () => {
+      const result = component.render();
+
+      expect(result).to.be.an.instanceof(TemplateResult);
+    });
+
+    it('should call the additionalInfo function', () => {
+      const additionalInfo = stub(component, 'additionalInfo');
+
+      component.render();
+
+      expect(additionalInfo).to.be.called;
+    });
+
+    it('should call the urlWrap function', () => {
+      const urlWrap = stub(component, 'urlWrap');
+
+      component.render();
+
+      expect(urlWrap).to.be.called;
+    });
+  });
+});
+
+describe('Variant Component', () => {
+  let component;
+
+  beforeEach(() => {
+    component = new Variant();
+  });
+
+  describe('constructor', () => {
+    it('should extend the Base class', () => {
+      expect(component).to.be.an.instanceOf(Base);
+    });
+
+    describe('type property', () => {
+      it('should have a default value of "text"', () => {
+        expect(component.type).to.equal('text');
+      });
+    });
+
+    describe('variant property', () => {
+      it('should default to an object with a text string and an empty product object', () => {
+        expect(component.variant.text).to.equal('');
+        expect(component.variant.product).to.deep.equal({});
+      });
+    });
+  });
+
+  describe('connectedCallback', () => {
+    it('should call its super connectedCallback', () => {
+      const connectedCallback = stub(Base.prototype, 'connectedCallback');
+
+      component.connectedCallback();
+
+      expect(connectedCallback).to.have.been.called;
+    });
+
+
+    it('should set role to "listitem"', () => {
+      const setAttribute = stub(component, 'setAttribute');
+
+      component.connectedCallback();
+
+      expect(setAttribute).to.have.been.calledWith('role', 'listitem');
+    });
+
+    describe('variant types', () => {
+      beforeEach(() => {
+        component.variant = {
+          text: 'Foo',
+          color: '#bed', // 'rgb(187, 238, 221)'
+          image: 'src.png',
+          product: {}
+        };
+      });
+
+      it('should set title and style if the type is "color"', () => {
+        component.type = 'color';
+
+        component.connectedCallback();
+
+        expect(component.style.backgroundColor).to.equal('rgb(187, 238, 221)');
```

Comment:
If we're going to be using this rgb value multiple times in the tests, I think it's worth it to put into a constant. Consider also replacing `#bed` with this, because both are valid colours.